pkg/controller/kccmanager/kccmanager.go

```diff
@@ -83,19 +82,19 @@ func New(ctx context.Context, restConfig *rest.Config, config Config) (manager.M
 		// the calls to AddToScheme(..) will modify the internal maps
 		opts.Scheme = runtime.NewScheme()
 	}
-	// Disable the cache. The cache causes problems in namespaced mode when trying
-	// to read resources in our system namespace.
-	opts.NewClient = nocache.NoCacheClientFunc
 	opts.BaseContext = func() context.Context {
 		return ctx
 	}
+	if err := addSchemes(opts.Scheme); err != nil {
+		return nil, fmt.Errorf("error adding schemes: %w", err)
+	}
+
+	// only cache CC and CCC resources
+	opts.Cache.ByObject = nocache.ByCCandCCC
```

Comment:
IIUC, we don't want to slam the api server for these resources, especially since we will attempt to fetch CC/CCCs for every CR's reconciliation that is managed by kcc.

Reply:
Gotcha, thank you!